### ISSUE TITLE
Snapping an interactable to SnapZone when they are both in same location breaks object following

### ISSUE THREAD

**Issue by Scraphead** [ExtendRealityLtd/Tilia.Interactions.SnapZone.Unity#84]
### Environment

* Code version: 
"io.extendreality.tilia.interactions.snapzone.unity": "1.1.43",
"io.extendreality.tilia.interactions.interactables.unity": "1.15.6",
"io.extendreality.tilia.mutators.objectfollower.unity": "1.1.13",
* Platform version: Unity 2019.4.19f, Windows 10
* 3rd party dependencies: None

### Steps to reproduce

1: Set position and rotation of InteractableFacade to match SnapZoneFacade postion and rotation.
2: Snap Interactable to SnapZone (In code)
3: Move SnapZoneFacade GameObject.
4: Interactable facade does not follow snapzone movement, even if it was snapped in code.

Code example
```
//Interactable facade with primary grab action follow, and secondary grab action swap
public InteractableFacade interactableFacade;

// SnapZone transition duration is the default 0
public SnapZoneFacade snapZoneFacade;

private void Start()
{
    //Snapping intractable to same location as SnapZone breaks Snapping Logic
    interactableFacade.transform.position = snapZoneFacade.transform.position;
    interactableFacade.transform.rotation = snapZoneFacade.transform.rotation;
    
    //Enable this line to make snapping work as intended
    // interactableFacade.transform.position = snapZoneFacade.transform.position + new Vector3(0, 0.1f, 0);
    
    //SnapZone now register an object snapped, but its Object follower moment processor does not enable.
    snapZoneFacade.Snap(interactableFacade);
}

private void Update() //Moving SnapZone
{
    var pos = snapZoneFacade.transform.position;
    pos.x += Time.deltaTime;
    snapZoneFacade.transform.position = pos;
}
```
![Imgur](https://imgur.com/wc671Mz.gif)

### Expected behavior

Interactable should follow SnapZone it is in

### Current behavior

Interactable does not follow SnapZone movements

**Comment by thestonefox** [comment]
I don't understand the steps to reproduce. what does step 1 mean?

**Comment by thestonefox** [comment]
Are you moving the transform to the same position and then snapping it? try offsetting the position before snapping it?

**Comment by Scraphead** [comment]
> 
> 
> I don't understand the steps to reproduce. what does step 1 mean?

More detailed steps.
1: Create a new unity project with tillia.
2: Create a Interactable and SnapZone from tillia prefabs. Place them next to each others.
3: Set up interactable facade with primary grab action follow, and secondary grab action swap. (To avoid errors)
4: Set interactable gameobject rigidbody (Is Kinematic) to true. (Only to keep it from falling)
6: Create a new script that can have a reference to SnapZoneFacade and InteractableFacade
7: Add to a script a Start() method that moves interactable facade world position and rotation match snapzone facade world position and rotation.
8: In same Start() method , after moving interactable, snap the interactable to snapzone with snapZoneFacade.Snap(interactableFacade);
9: Add script to scene on a new gameobject, with references to scene interactable and facade.
10: Press play.
11: You should see interactable move to snapzone, it should be snapped, (And it is when checking out snapped items in snapzone)
12: Grab the gameobject of snapzone facade in scene view and move it around. Interactable should follow snapzone facade, but it does not.

**Comment by Scraphead** [comment]
> 
> 
> Are you moving the transform to the same position and then snapping it? try offsetting the position before snapping it?

Yes, that works. My current workaround is spawning it with a slight offset in rotation. (That works too) The issue was during runtime I spawned a new Interactable at the same location as snapzone, and got confused when it did not follow snapzone. The fix by not spawning it right on top of snapzone is a simple one, but not expected behavior.

**Comment by thestonefox** [comment]
Try this fix: https://github.com/ExtendRealityLtd/Tilia.Interactions.SnapZone.Unity/pull/89

**Comment by Scraphead** [comment]
> 
> 
> Try this fix: #89

Can confirm that issue is fixed in demo scene I had set up. Great!
